chore(core): move compiler out of core

vsavkin opened the pull request:
BREAKING CHANGE

All imports from 'angular2/core/compiler' should be changed to 'angular2/compiler'.

mary-poppins commented:
Merging PR #5147 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5147](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5147).

mary-poppins commented:
Merging PR #5147 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5147](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5147).

vsavkin committed:
chore(core): move compiler out of core

BREAKING CHANGE

All imports from 'angular2/core/compiler' should be changed to 'angular2/compiler'.

vsavkin commented:
Closed via https://github.com/angular/angular/commit/6edd964a83fb7c29dc82cb0731959a799e2fbde1
